main.go

```diff
@@ -19,50 +20,63 @@ import (
 	"github.com/almighty/almighty-core/remoteworkitem"
 	"github.com/almighty/almighty-core/transaction"
 	token "github.com/dgrijalva/jwt-go"
+	"github.com/fsnotify/fsnotify"
 	"github.com/goadesign/goa"
 	"github.com/goadesign/goa/middleware"
 	"github.com/goadesign/goa/middleware/security/jwt"
+	"github.com/spf13/viper"
+	yaml "gopkg.in/yaml.v2"
 )
 
 var (
 	// Commit current build commit set by build script
 	Commit = "0"
 	// BuildTime set by build script
 	BuildTime = "0"
-	// Development enables certain dev only features, like auto token generation
-	Development = false
-)
-
-const (
-	// DBMaxRetryAttempts is the number of times alm server will attempt to open a connection to the database before it gives up
-	DBMaxRetryAttempts int = 50
 )
 
 func main() {
-	printUserInfo()
-
-	var dbHost string
+	// --------------------------------------------------------------------
+	// Parse flags
+	// --------------------------------------------------------------------
+	var configFilePath string
+	var printConfig bool
 	var scheduler *remoteworkitem.Scheduler
-
-	flag.BoolVar(&Development, "dev", false, "Enable development related features, e.g. token generation endpoint")
-	flag.StringVar(&dbHost, "dbhost", "", "The hostname of the db server")
+	flag.StringVar(&configFilePath, "config", "config.yaml", "Path to the config file to read")
+	flag.BoolVar(&printConfig, "printConfig", false, "Prints the config (including merged environment variables) and exits")
 	flag.Parse()
 
-	if len(dbHost) == 0 {
-		dbHost = os.Getenv("DBHOST")
+	var err error
+	if err = setupConfiguration(configFilePath); err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
 	}
 
-	if len(dbHost) == 0 {
-		dbHost = "localhost"
+	if printConfig {
+		allSettings := viper.AllSettings()
+		var y []byte
+		y, err = yaml.Marshal(&allSettings)
+		if err != nil {
+			panic(fmt.Errorf("Failed to marshall config to string: %s", err.Error()))
+		}
+		fmt.Printf("%s\n", y)
+		os.Exit(0)
 	}
 
+	printUserInfo()
```

Comment:
Hi @baijum . No worries. Take a look at the [function](https://github.com/almighty/almighty-core/blob/1efe0b9f0f10f3c751215a7637405ac555482e92/main.go#L155). It just prints the Username and the user's ID that is currently running the binary. The idea behind this is to more easily see if there might be user errors when running almighty inside a container.